### PR TITLE
fix: address PR comments from #211

### LINKED ISSUE #223: Address comments in PR #211
- https://github.com/hashgraph/solo/pull/211#discussion_r1559052174
- https://github.com/hashgraph/solo/pull/211#discussion_r1559054567
- https://github.com/hashgraph/solo/pull/211#discussion_r1559055542

### PR DESCRIPTION
**Pull request by jeromy-cannon**
## Description

This pull request changes the following:

- correct a typo in `node.mjs`
- update refresh command description
- add jsdoc for `helper.mjs` `validatePath()`
- updates to `flow-update-readme.yaml` to make when it runs more consistent and fix:
  - GITHUB_TOKEN is not signing commits
  - updated to only run when pushes occur to `main` branch

### Related Issues

* Closes #223

### CODE REVIEW

**Review by JeffreyDallas — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Review by nathanklick — COMMENTED** [review]
@jeromy-cannon Let's sync up on Monday regarding this PR. I need some help understanding some of the changes and how they differ from the previous iteration.

File: .github/workflows/flow-update-readme.yaml [hunk 1]
```diff
@@ -157,14 +188,29 @@ jobs:
...
-      # Signed Commits by actions https://github.com/actions/runner/issues/667
       - name: Commit README.md Changes
-        if: ${{ steps.check-readme-changes.outputs.TOTAL_LINES_CHANGED > 20 }}
-        uses: planetscale/ghcommit-action@v0.1.35
+        if: ${{ steps.readme-update-check.outputs.run-update }}
+        uses: stefanzweifel/git-auto-commit-action@8621497c8c39c72f3e2a999a26b4ca1b5058a842 # v5.0.1
         with:
           commit_message: "auto update README.md"
-          repo: ${{ github.repository }}
           branch: ${{ github.head_ref || github.ref_name }}
```
Comment (jeromy-cannon) [inline .github/workflows/flow-update-readme.yaml:207]:
```suggestion
```

File: .github/workflows/flow-update-readme.yaml [hunk 2]
```diff
@@ -50,14 +43,46 @@ jobs:
         with:
           fetch-depth: 0
           ref: ${{ github.event.pull_request.head.ref }}
```
Comment (jeromy-cannon) [inline .github/workflows/flow-update-readme.yaml:45]:
```suggestion
```

File: .github/workflows/flow-update-readme.yaml [hunk 3]
```diff
@@ -50,14 +43,46 @@ jobs:
...
+          echo "# determine if commits since last 'auto update README.md' commit"
+          git --no-pager log --reverse --pretty="%H|%an|%s" origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.ref }} || true
+          COMMITS_SINCE_UPDATE=$(git --no-pager log --reverse --pretty="%H|%an|%s" origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.ref }} | awk '/Swirlds Automation\|auto update README\.md/ {s=""; next}  {s = s $0 RS} END {printf "%s", s}') || true
+          [[ "${#COMMITS_SINCE_UPDATE}" -gt 0 ]] && NEW_COMMITS=true || NEW_COMMITS=""
+          echo "NEW_COMMITS=$NEW_COMMITS"
+
+          echo "# determine if we need to update the README.md"
+          [[ "${FILES_CHANGED}" == "true" && "${NEW_COMMITS}" == "true" ]] && RUN_UPDATE=true || RUN_UPDATE=""
+          echo "RUN_UPDATE=$RUN_UPDATE"
+          echo "run-update=$RUN_UPDATE" >> "${GITHUB_OUTPUT}"
```
Comment (jeromy-cannon) [inline .github/workflows/flow-update-readme.yaml:76]:
```suggestion
```

**Pull request merged**